olp-cpp-sdk-dataservice-read/tests/StreamLayerClientImplTest.cpp

```diff
@@ -718,38 +703,309 @@ TEST(StreamLayerClientImplTest, GetDataCancel) {
   EXPECT_EQ(response.GetError().GetErrorCode(), ErrorCode::Cancelled);
 }
 
+TEST(StreamLayerClientImplTest, Poll) {
+  auto network_mock = std::make_shared<NetworkMock>();
+  auto cache_mock = std::make_shared<CacheMock>();
+  OlpClientSettings settings;
+  settings.network_request_handler = network_mock;
+  settings.cache = cache_mock;
```

Comment:
Now after 10+ test cases I don't think it was worth it to use TEST instead of TEST_F with setting up this entire stuff once?
Please change and create a simple fixture that sets up this entire stuff once and possibly also creates a StreamLayerClient for you.